target_chains/ethereum/sdk/solidity/MockPyth.sol

```diff
@@ -160,6 +173,186 @@ contract MockPyth is AbstractPyth {
             );
     }
 
+    function parseTwapPriceFeedUpdates(
+        bytes[][] calldata updateData,
+        bytes32[] calldata priceIds
+    )
+        external
+        payable
+        returns (PythStructs.TwapPriceFeed[] memory twapPriceFeeds)
+    {
+        // Validate inputs and fee
+        if (updateData.length != 2) revert PythErrors.InvalidUpdateData();
+
+        uint requiredFee = getUpdateFee(updateData[0]);
+        if (requiredFee != getUpdateFee(updateData[1]))
+            revert PythErrors.InvalidUpdateData();
+        if (msg.value < requiredFee) revert PythErrors.InsufficientFee();
+
+        twapPriceFeeds = new PythStructs.TwapPriceFeed[](priceIds.length);
+
+        // Process each price ID
+        for (uint i = 0; i < priceIds.length; i++) {
+            processTwapPriceFeed(updateData, priceIds[i], i, twapPriceFeeds);
+        }
+    }
+
+    function processTwapPriceFeed(
```

Comment:
what you are doing in this method is strange. you should just have a create twap price function that gives you binary and parse it here; parsing pricefeed converting it to cumulative price and then calculating twap is a unnecessary.